packages/@aws-cdk/cloudformation-include/test/yaml-templates.test.ts

```diff
@@ -0,0 +1,382 @@
+                    ]],
+                  },
+                  "StorageClass": "STANDARD",
+                },
+                "Id": "Backup",
+                "Prefix": "",
+                "Status": "Enabled",
+              }],
+            },
+            "VersioningConfiguration": {
+              "Status": "Enabled",
+            },
+          },
+        },
+        "WorkItemBucketBackupRole": {
+          "Type": "AWS::IAM::Role",
+          "Properties": {
+            "AssumeRolePolicyDocument": {
+              "Statement": [{
+                "Action": ["sts:AssumeRole"],
+                "Effect": "Allow",
+                "Principal": {
+                  "Service": ["s3.amazonaws.com"],
+                },
+              }],
+            },
+          },
+        },
+        "BucketBackupPolicy": {
+          "Type": "AWS::IAM::Policy",
+          "Properties": {
+            "PolicyDocument": {
+              "Statement": [{
+                "Action": [
+                  "s3:GetReplicationConfiguration",
+                  "s3:ListBucket",
+                ],
+                "Effect": "Allow",
+                "Resource": [{
+                  "Fn::Join": ["", [
+                    "arn:aws:s3:::", {
+                      "Ref": "RecordServiceS3Bucket",
+                    },
+                  ],
+                  ],
+                }],
+              }, {
+                "Action": [
+                  "s3:GetObjectVersion",
+                  "s3:GetObjectVersionAcl",
+                ],
+                "Effect": "Allow",
+                "Resource": [{
+                  "Fn::Join": ["", [
+                    "arn:aws:s3:::", {
+                      "Ref": "RecordServiceS3Bucket",
+                    },
+                    "/*",
+                  ]],
+                }],
+              }, {
+                "Action": [
+                  "s3:ReplicateObject",
+                  "s3:ReplicateDelete",
+                ],
+                "Effect": "Allow",
+                "Resource": [{
+                  "Fn::Join": ["", [
+                    "arn:aws:s3:::", {
+                      "Fn::Join": ["-", [
+                        { "Ref": "AWS::Region" },
+                        { "Ref": "AWS::StackName" },
+                        "replicationbucket",
+                      ]],
+                    },
+                    "/*",
+                  ]],
+                }],
+              }],
+            },
+            "PolicyName": "BucketBackupPolicy",
+            "Roles": [{
+              "Ref": "WorkItemBucketBackupRole",
+            }],
+          },
+        },
+      },
+    });
+  });
+
+  test('can ingest a yaml template with parameters and output it unchanged', () => {
+    includeTestTemplate(stack, 'bucket-with-parameters.yaml');
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject('bucket-with-parameters.yaml'),
+    );
+  });
+});
+
+interface IncludeTestTemplateProps {
```

Comment:
A valiant effort, but just forget about this for now :). Just have `includeTestTemplate` take 1 parameter.